drivers/sensor/mhz19b/mhz19b.c

```diff
@@ -0,0 +1,324 @@
+/*
+ * Copyright (c) 2021 G-Technologies Sdn. Bhd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "mhz19b-priv.h"
+
+#if !IS_ENABLED(CONFIG_UART_INTERRUPT_DRIVEN)
+#error UART_INTERRUPT_DRIVEN is required
+#endif
+
+static void mhz19b_uart_flush(const struct device *uart_dev)
+{
+	uint8_t c;
+
+	while (uart_fifo_read(uart_dev, &c, 1) > 0) {
+		continue;
+	}
+}
+
+static uint8_t mhz19b_checksum(const uint8_t *data)
+{
+	uint8_t cs = 0;
+
+	for (uint8_t i = 1; i < MHZ19B_BUF_LEN - 1; i++) {
+		cs += data[i];
+	}
+
+	return 0xff - cs + 1;
+}
+
+static int mhz19b_send_cmd(const struct device *dev, const enum mhz19b_cmd cmd, const bool has_rsp)
+{
+	struct mhz19b_data *data = dev->data;
+	const struct mhz19b_cfg *cfg = dev->config;
+	int ret;
+
+	/* Make sure last command has been transferred */
+	ret = k_sem_take(&data->tx_sem, MHZ19B_WAIT);
+	if (ret) {
+		return ret;
+	}
+
+	data->cmd = cmd;
+	data->has_rsp = has_rsp;
+	k_sem_reset(&data->rx_sem);
+
+#ifdef CONFIG_MHZ19B_KEEP_MCU_AWAKE
+	if (has_rsp) {
+		pm_constraint_set(PM_STATE_SUSPEND_TO_IDLE);
+	}
+#endif
+
+	uart_irq_tx_enable(cfg->uart_dev);
+
+	if (has_rsp) {
+		uart_irq_rx_enable(cfg->uart_dev);
+		ret = k_sem_take(&data->rx_sem, MHZ19B_WAIT);
+#ifdef CONFIG_MHZ19B_KEEP_MCU_AWAKE
+		pm_constraint_release(PM_STATE_SUSPEND_TO_IDLE);
+#endif
+	}
+
+	return ret;
+}
+
+static inline int mhz19b_send_config(const struct device *dev, const enum mhz19b_cmd cmd)
+{
+	struct mhz19b_data *data = dev->data;
+	int ret;
+
+	ret = mhz19b_send_cmd(dev, cmd, true);
+	if (ret == 0) {
```

Comment:
What do you think about something like that:

<pre>
ret = mhz19b_send_cmd(dev, cmd, true);
if (ret < 0) {
	return ret;
}

if (data->rd_data[MHZ19B_RX_CMD_IDX] != mhz19b_cmds[data->cmd][MHZ19B_TX_CMD_IDX]) {
	return -EINVAL;
}

return 0;
</pre>

Reply:
Fixed, thanks